### PR TITLE
ocamlPackages.mldoc: init at 1.3.9

### PR DESCRIPTION
**Pull request by marsam**
###### Description of changes

Add https://github.com/logseq/mldoc

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by vbgl — COMMENTED** [review]
It seems to require OCaml ≥ 4.10. Please set `minimalOCamlVersion` accordingly.

File: pkgs/development/ocaml-modules/mldoc/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    core
+    core_bench
+    js_of_ocaml
+    js_of_ocaml-ppx
+    ppx_deriving_yojson
+    uri'
+    yojson
+    lwt
+    xmlm
+  ];
```
[the remark this answers is not in the document]
Comment (marsam) [reply pkgs/development/ocaml-modules/mldoc/default.nix:57]: Sorry for the delay. You are right, those packages don't need to be propagated. Thanks!

[the remark this answers is not in the document]
Comment (vbgl) [reply pkgs/development/ocaml-modules/mldoc/default.nix:57]: No problem. I think that `ppx_deriving` is needed:

> $ ocamlfind query -r mldoc
ocamlfind: Package \`ppx_deriving.runtime' not found - required by `mldoc'

[the remark this answers is not in the document]
Comment (marsam) [reply pkgs/development/ocaml-modules/mldoc/default.nix:57]: oh, fixed. Thanks again

**Review by vbgl — APPROVED** [review]

**Pull request merged**